### ISSUE TITLE
(Neoforge 1.21.1) Shift-Right Click UI is invisible without OP/commands enabled.

### ISSUE THREAD

**Issue by ogpw** [jaquadro/StorageDrawers#1197]
Neoforge 21.1.90, Minecraft 1.21.1, Java 21.0.5.11 Eclipse Adoptium

Storage Drawers 13.8.5 + other mods.

I have only tested singleplayer, however, someone else mentions the same issue on their multiplayer issue (but for fabric but still 1.21).

Issue:
Shift-right-clicking storage drawer opens an invisible GUI (can still hover over items and get tooltip text of stored items and upgrades) instead of opening the normal drawer GUI.

Re-creating issue:
Create a new singleplayer world WITHOUT cheats/commands enabled. Place a storage drawer and attempt to right-click. Voila.

Workaround to issue:
Either create a new singleplayer world w/ cheats/commands enabled.
OR
w/ minecraft open, edit the level.dat with NBTExplorer (or another NBT editor) and change allowCommands from 0 to 1. load world. shift-right-click still doesn't work. exit and save world, quit game. restart game. load up world. shift-right-click should now work as normal (no invisible GUI).

Modlist:
- Alternate Current
- Architectury
- Ars Nouveau
- Ars Nouveau's Flavors & Delight
- Better With Minecolonies
- Building Gadgets 2
- CC: Tweaked
- Caelus API
- Chunky
- City Craft
- Cloth Config v15 API
- Concurrent Chunk Management Engine
- Continents
- CorgiLib
- Cucumber Library
- Curios API
- Dis\-Enchanting Table
- Distant Horizons
- Do a Barrel Roll
- Domum Ornamentum
- Dynamic FPS
- Easy Anvils
- Easy Magic
- Elytra Slot
- EnchantmentLevelBreak
- Enhanced Celestials
- FTB Essentials
- FTB Library
- FTB Ultimine
- Farmer's Delight
- Ferrite Core
- Forgified Fabric API
- GeckoLib 4
- HammerLib
- ImmediatelyFast
- Iris
- Iron Chests
- Iron Furnaces
- Iron Jetpacks
- Iron Ladders
- Iron's Spells 'n Spellbooks
- Just Dire Things
- Lithium
- McJtyLib
- Merge Enchantments
- MineColonies
- Mob Grinding Utils
- ModernFix
- MonoLib
- More Red
- More Red x CC:Tweaked Compat
- Mouse Tweaks
- Multi\-Piston
- Ok Zoomer
- ParCool\!
- Patchouli
- Player Animator
- Polymorph
- Puzzles Lib
- RFToolsBase
- RFToolsBase
- RFToolsBuilder
- RFToolsPower
- Roughly Enough Items \(REI\)
- Sodium
- Solar Flux Reborn
- Sophisticated Backpacks
- Sophisticated Core
- Storage Delight
- Storage Drawers
- Structurize
- SuperMartijn642's Config Library
- SuperMartijn642's Core Lib
- Tax Free Levels
- Tectonic
- Tom's Peripherals
- Torcherino
- TownTalk
- UI Library Mod
- Ultimate Car Mod
- Ultimate Plane Mod
- Wormhole \(Portals\)
- XNet
- Xaero's Minimap
- Xaero's World Map

**Comment by Jack-McKalling** [comment]
It seems I cannot reproduce this bug with just Storage Drawers installed. I suspect this issue is caused by one of the mods you listed.

**Comment by IsThatSkaR** [comment]
Same issue here. Playing on local coop with e4mc, my friends can not see the UI.

Your workaround does seem to work.

**Comment by benbenlaw** [comment]
seems to be when ftb libaray is installed

**Comment by Jack-McKalling** [comment]
> seems to be when ftb libaray is installed

Yes indeed, I found out yesterday on discord too from someone else's crashlog.
The FTB library is giving a mipmap warning about the `icons/dropdown_out` texture. Its size is only 15x15 pixels, which means mipmapping is limited and that might be why some of our block state definitions cannot be loaded (it says that a lot of our models are missing).

**Comment by benbenlaw** [comment]
at least you know the issue haha

**Comment by benbenlaw** [comment]
so is this an issue on the ftb side or storage drawer side?

**Comment by Jack-McKalling** [comment]
It's not clear yet whether the FTB error is just a coincidence or indeed the cause. Texel said on the discord that the meta model errors can just be ignored, but there were some others, and he mentioned that he'll look into those soon.

I suspect that there could be some tweaking or fixing on both sides, but I'd want to await the result to see what's really happening before I start pointing an actual finger.

**Comment by benbenlaw** [comment]
i think its ftb after some digging into the libaray 

![Image](https://github.com/user-attachments/assets/064cc4d8-064f-40fe-a532-52515be3d8fc)

setting this to false allows it to open

**Comment by benbenlaw** [comment]
so i have implemented a fix using fancy menu if anyone needs this in the future (encase it doesnt get fixed)

**Comment by MkGrinngo** [comment]
I'm have some players on my 1.21 fabric server with this same issue. Ftb settings didn't fix it. 😔

**Comment by Jack-McKalling** [comment]
> I'm have some players on my 1.21 fabric server with this same issue. Ftb settings didn't fix it. 😔

That's a different modloader, a lot of stuff works differently there. If something was fixed in neoforge, it also has to specifically get ported to fabric too, before it's fixed there as well.

**Comment by celestial-cookie** [comment]
Having the same issue on our 1.21.1 Neoforge server, and FTB settings didn't fix it unfortunately.

**Comment by ResistorCat** [comment]
Same here, fabric 1.21.1 on server not working for non OP players

**Comment by jaquadro** [comment]
I was able to reproduce this in dev client/server with FTB Library present, so hopefully can get to the bottom of it.

Client/server without FTB Library seems okay.  Client-only with FTB Library also seems okay.  The buttons setting made no difference.

**Comment by jaquadro** [comment]
Think I've identified the fix.  The next releases on 1.21.1 should include it.

**Comment by jaquadro** [comment]
This should be resolved in the 13.8.10 release now
